src/tracing.rs

```diff
@@ -57,14 +60,25 @@ where
 pub fn open_telemetry<S>(
     service_name: &'static str,
     service_version: &'static str,
-) -> Result<impl Layer<S>, TraceError>
+) -> Result<Box<dyn Layer<S>>, TraceError>
 where
     S: Subscriber + for<'span> LookupSpan<'span>,
 {
+    let disabled = env::var("OTEL_SDK_DISABLED").ok().unwrap_or_default() == "true";
+
+    if disabled {
```

Comment:
You can compare `Option<T>` directly without unwrapping (as_deref could be required due to string/str)